src/arrayMethodMap.js

```diff
@@ -5,7 +5,11 @@
  */
 function applyCustomMap() {
   [].__proto__.map2 = function(callback) {
-    // write code here
+    let result = [];
+    this.filter((el, i) => {
```

Comment:
Please avoid using `filter`, `map` when you don't need to return any value from the loop. It's like using a screwdriver to stir a tea. It works but ...